src/components/Map/Legend/index.jsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styles from './styles.scss';
+
+const propTypes = {
+    className: PropTypes.string,
+    legendItems: PropTypes.arrayOf(PropTypes.shape({
+        label: PropTypes.string,
+        color: PropTypes.color,
+    })),
+    onStateChange: PropTypes.func,
+};
+
+const defaultProps = {
+    className: '',
+    legendItems: [],
+    fontSize: 12,
+    onStateChange: undefined,
+};
+
+export default class Legend extends React.PureComponent {
+    static propTypes = propTypes;
+    static defaultProps = defaultProps;
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            inactive: [],
+            maxSize: Math.max(...props.legendItems.map(item => item.size || 10)),
```

Comment:
If size is 0, should it be replaced with 10?
or size is never 0?

Reply:
This is a useless thing from idmc. We should remove it entirely. :yum: